thunder/core/module.py

```diff
@@ -141,6 +143,101 @@ def load_original_state_dict(self, state_dict):
                 else:
                     raise NotImplementedError(f"don't know how to handle {full_k}")
 
+    def state_dict(self, *, destination=None, prefix="", keep_vars=False):
+        """
+        Save state dict of the (transformed) Thunder module.
+        Note that this is rather more rudimentary than the original state_dict.
+        """
+        if destination is None:
+            destination = collections.OrderedDict()
+            destination._metadata = collections.OrderedDict()
+
+        for name, param in self.named_parameters(prefix=prefix):
+            if param is not None:
+                destination[name] = param if keep_vars else param.detach()
+
+        non_persistent_buffers_set = set()
+        for name, submodule in self._model.named_modules(prefix=prefix):
+            subprefix = f"{name}." if name else name
+            non_persistent_buffers_set.update(f"{subprefix}{bname}" for bname in submodule._non_persistent_buffers_set)
+
+        for name, buf in self.named_buffers(prefix=prefix):
+            if buf is not None and name not in non_persistent_buffers_set:
+                destination[name] = buf if keep_vars else buf.detach()
+
+        for name, submodule in self._model.named_modules(prefix=prefix):
+            subprefix = f"{name}." if name else name
+            extra_state_key = subprefix + pytorch.nn.modules.module._EXTRA_STATE_KEY_SUFFIX
+            if (
+                getattr(self.__class__, "get_extra_state", pytorch.nn.Module.get_extra_state)
+                is not pytorch.nn.Module.get_extra_state
+            ):
+                destination[extra_state_key] = self.get_extra_state()
+        return destination
+
+    def load_state_dict(self, state_dict: Mapping[str, Any], strict: bool = True, assign: bool = False):
+        # this is much more simple than the original (hooks etc.)
+
+        # non-persistent buffer overrides?
+        non_persistent_buffers_set = set()
+        for name, submodule in self._model.named_modules():
+            subprefix = f"{name}." if name else name
+            non_persistent_buffers_set.update(f"{subprefix}{bname}" for bname in submodule._non_persistent_buffers_set)
+
+        keys_unused = thunder.core.utils.OrderedSet(state_dict.keys())
+        keys_missing = thunder.core.utils.OrderedSet()
+        errors = []
+        for name, v in itertools.chain(self.named_parameters(), self.named_buffers()):
+            if name not in non_persistent_buffers_set:
+                if name not in state_dict:
```

Comment:
super minor nit: maybe we can continue on `name in non_persistent_buffers_set` and save a level of indentation?